modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -16,36 +16,42 @@ def create_a_record(fqdn, ip)
       if found = dns_find(fqdn)
         raise(Proxy::Dns::Collision, "#{fqdn} is already used by #{found}") if found != ip
       else
-        zone = match_zone(fqdn)
+        zone = match_zone(fqdn, enum_zones)
         msg = "Added DNS entry #{fqdn} => #{ip}"
         cmd = "/RecordAdd #{zone} #{fqdn}. A #{ip}"
         execute(cmd, msg)
       end
     end
 
-    # noop
-    def create_ptr_record(fqdn, ip)
-      found = dns_find(ip)
-      raise(Proxy::Dns::Collision, "#{ip} is already used by #{found}") if found && found != fqdn
-      true
+    def create_ptr_record(fqdn, ptr)
+      ip = ptr_to_ip ptr
+      if found = dns_find(ip)
+        raise(Proxy::Dns::Collision, "#{ptr} is already used by #{found}") unless fqdn == found
+      else
+        zone = match_zone(ptr, enum_zones)
+        msg = "Added PTR entry #{ptr} => #{fqdn}"
+        cmd = "/RecordAdd #{zone} #{ptr}. PTR #{fqdn}."
+        execute(cmd, msg)
+      end
     end
 
     def remove_a_record(fqdn)
```

Comment:
You mean ```dns_find```, right? There is indeed an issue with it -- if fqdn or ip has multiple mappings, ```Resolv::DNS``` will round-robin (I believe) return them. Changes in ```dns_find``` will affect all modules though, I'd rather do that in a dedicated PR.